src/server.ts

```diff
@@ -0,0 +1,22 @@
+import * as path from 'path'
+import * as express from 'express'
+import * as Logger from 'bunyan'
+// Teach express to properly handle async errors
+require('express-async-errors')
+
+import {logRequest} from './middleware/logging'
+
+export const createServer = function (args: ServerArgs) {
+  const app:express.Application = express()
+
+  app.use(logRequest({logger: args.logger}))
+  app.use('/probot/static/', express.static(path.join(__dirname, '..', 'static')))
+  app.use(args.webhook)
+  app.set('view engine', 'hbs')
+  app.set('views', path.join(__dirname, '..', 'views'))
+  app.get('/ping', (req, res) => res.end('PONG'))
+
+  return app
+}
+
+export interface ServerArgs {webhook: express.Application, logger: Logger}
```

Comment:
TypeScript best-practice question…

When do you export an interface for function parameters vs just declaring them inline? Could the `createServer` function above be declared like this?

```ts
export const createServer = function ({webhook: express.Application, logger: Logger}) {
```

Reply:
In this instance, if you try and define it inline, everything in between `{}` gets treated as key/value pairs, but `express.Application` is a type, not a value. One of the things I've struggled with is that things between the brackets are sometimes objects and sometimes interfaces. This may be a good reason for defining your types in separate files, honestly. It may help differentiate between the two.